lib/Doctrine/ORM/Mapping/Driver/AttributeReader.php

```diff
@@ -14,58 +15,93 @@
 use function assert;
 use function is_string;
 use function is_subclass_of;
+use function sprintf;
 
 /**
  * @internal
  */
 final class AttributeReader
 {
-    /** @var array<string,bool> */
+    /** @var array<class-string<Annotation>,bool> */
     private array $isRepeatableAttribute = [];
 
-    /** @return array<Annotation|RepeatableAttributeCollection> */
+    /**
+     * @return class-string-map<T, T|RepeatableAttributeCollection<T>>
```

Comment:
I didn't know `class-string-map` – which tool supports that type?

Also, I think we usually use a vendor-prefixed annotation like `@psalm-return` for types that not every tool might understand.

Reply:
This is supported by Psalm: https://github.com/vimeo/psalm/search?q=class-string-map
It does not seem supported by PHPStan (I could not find an occurrence in the codebase).